src/BuildScriptGenerator/DefaultBuildScriptGenerator.cs

```diff
@@ -292,26 +292,7 @@ private void LogScriptIfGiven(string type, string scriptPath)
         /// <returns>The maximum version that satisfies the requested version spec.</returns>
         private string GetMatchingTargetVersion(IProgrammingPlatform platform, string targetVersionSpec)
         {
-            string targetVersion;
```

Comment:
This piece of logic has been removed because:
1. This duplicates with what each individual platform's detector does.
2. SemanticVersionResolver does not handle each platform's preview format resolution. For example currently it does not handle .NET Core preview version format. I am fixing this issue in .NET Core's platform detector instead of having switch-case conditions within SemanticVersionResolver. Now since this detector handles preview version too, I am removing this code and pushing this responsibility to each individual platform itself.